components/script/dom/htmlinputelement.rs

```diff
@@ -1075,20 +1057,25 @@ impl HTMLInputElement {
         TextControlSelection::new(&self, &self.textinput)
     }
 
-    fn update_text_contents(&self, value: DOMString, update_text_cursor: bool) -> ErrorResult {
+    fn update_text_contents(&self, mut value: DOMString, update_text_cursor: bool) -> ErrorResult {
         match self.value_mode() {
             ValueMode::Value => {
-                // Steps 1-2.
-                let old_value = mem::replace(self.textinput.borrow_mut().single_line_content_mut(), value);
                 // Step 3.
                 self.value_dirty.set(true);
+
                 // Step 4.
                 if update_text_cursor {
-                    self.sanitize_value();
+                    self.sanitize_value(&mut value);
```

Comment:
No need to check `update_text_cursor` any more.

Reply:
Done (although that's not really specific to this change, since `update_text_cursor` is/was always true here)